src/clsourcedependencies_parser.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NINJA_CLSOURCEDEPENDENCIESPARSER_H_
+#define NINJA_CLSOURCEDEPENDENCIESPARSER_H_
+
+#include <set>
+#include <string>
+
+/// This class parses the output of of the /sourceDependencies flag from Visual
+/// Studio's cl.exe, which generates a JSON file with dependency information.
+struct CLSourceDependenciesParser {
+  /// Parse the JSON file produced by /sourceDependencies, filling includes_
+  /// with the includes. Returns true on success, or false with err filled.
+  bool Parse(const std::string& content, std::string* err);
```

Comment:
Use string-piece instead of std::string const&

Reply:
Changed to use StringPiece.